### PR TITLE
Cleanup Node Layout: fix on groups and remove left alignment

### PR DESCRIPTION
**Pull request by ellensi**
### Purpose

This PR is to resolve the following two issues:

##### [MAGN-8271](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8271) Groups should be horizontally aligned when they all are from same graph
image
This is to fix a bug from the previous pull request where groups became laid out vertically if nodes from a group were connected to nodes in a different group.

##### [MAGN-8299](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8299) Input nodes should not be aligned to the left
image
This is done in order to minimize the overall connector length and the number of crossings.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@Benglin 

### FYIs

@riteshchandawar

### CODE REVIEW

**Comment by riteshchandawar** [comment]
This looks awesome now. 
@Racel Please provide your comments.

Thanks,
Ritesh

**Comment by Racel** [comment]
@riteshchandawar - I agree. It looks great!!! LGTM!

File: src/Engine/GraphLayout/GraphLayout.cs [hunk 1]
```diff
@@ -580,15 +573,6 @@ public class Node
...
 
-        /// <summary>
-        /// Gets a Boolean value of whether the node has any unconnected input ports.
-        /// </summary>
-        /// <value>True if the node has at least one unconnected input port, otherwise false.</value>
-        public bool HasUnconnectedInPort
-        {
-            get { return LeftEdges.Count < InPortCount; }
-        }
-
```
Comment (Benglin) [inline src/Engine/GraphLayout/GraphLayout.cs]: Dead code removal?

File: src/Engine/GraphLayout/GraphLayout.cs [hunk 2]
```diff
@@ -69,16 +68,16 @@ public Node FindNode(Guid guid)
...
         /// </summary>
         /// <param name="start">Start node.</param>
         /// <param name="end">End node.</param>
         /// <returns>The edge object.</returns>
         public Edge FindEdge(Node start, Node end)
         {
-            foreach (Edge edge in Edges.Where(x => x.Active))
+            foreach (Edge edge in Edges)
             {
-                if (start.Equals(edge.StartNode) && end.Equals(edge.EndNode))
```
Comment (Benglin) [inline src/Engine/GraphLayout/GraphLayout.cs]: Why is this change needed?

**Comment by Benglin** [comment]
The above `EngOps Build` failed for the following reason:

```
Auto-merging src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs
CONFLICT (content): Merge conflict in src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs
Automatic merge failed; fix conflicts and then commit the result.
```

I don't see anything wrong with your code changes, feel free to merge it in if you have time to wait for build results. Thanks for fixing this again it looks great!

File: src/Engine/GraphLayout/GraphLayout.cs (hunk 2 above)
Comment (ellensi) [inline src/Engine/GraphLayout/GraphLayout.cs]: The revised code is noticeably faster to run on large graphs.
